### PR TITLE
Mapped a bunch of villager and gossip related stuff

### PR DESCRIPTION
**Pull request by KaptainWutax**
I figured out gossips, iron golem mechanics and a few other misc methods in VillagerEntity.

### CODE REVIEW

**Comment by liach** [comment]
Just make sure this one does not conflict with #533

**Comment by KaptainWutax** [comment]
@liach Is there a quick way to check that? I'll go through them manually for now.

**Comment by liach** [comment]
hmm, i think you can try a compare between the 2 branches

File: mappings/anw.mapping [hunk 1]
```diff
@@ -0,0 +1,18 @@
+CLASS anw
+	FIELD a DUMMY Lanw;
```
Comment (Gegy) [inline mappings/anw.mapping]: This class has already been mapped, but Git does not detect the conflict.

File: mappings/anx.mapping [hunk 2]
```diff
@@ -0,0 +1,4 @@
+CLASS anx
```
Comment (Gegy) [inline mappings/anx.mapping]: This too.

File: mappings/aoo.mapping [hunk 3]
```diff
@@ -0,0 +1,7 @@
+CLASS aoo
```
Comment (Gegy) [inline mappings/aoo.mapping]: This too.

File: mappings/apb.mapping [hunk 4]
```diff
@@ -0,0 +1,18 @@
+CLASS apb
```
Comment (Gegy) [inline mappings/apb.mapping]: This too

File: mappings/net/minecraft/entity/SpawnType.mapping [hunk 5]
```diff
@@ -1 +1,17 @@
 CLASS aii net/minecraft/entity/SpawnType
+	FIELD a NATURAL Laii;
```
Comment (Gegy) [inline mappings/net/minecraft/entity/SpawnType.mapping]: If these names are inferred within Enigma, they shouldn't be mapped

File: mappings/net/minecraft/village/GossipType.mapping [hunk 6]
```diff
@@ -0,0 +1,22 @@
+CLASS anu net/minecraft/village/GossipType
+	FIELD a MAJOR_NEGATIVE Lanu;
```
Comment (Gegy) [inline mappings/net/minecraft/village/GossipType.mapping]: These are inferred within Enigma, and shouldn't be mapped

File: mappings/net/minecraft/village/InteractionType.mapping [hunk 7]
```diff
@@ -0,0 +1,6 @@
+CLASS aov net/minecraft/village/InteractionType
+	FIELD a ZOMBIE_VILLAGER_CURED Laov;
```
Comment (Gegy) [inline mappings/net/minecraft/village/InteractionType.mapping]: These too

File: mappings/net/minecraft/village/VillagerProfession.mapping [hunk 8]
```diff
@@ -1,3 +1,27 @@
 CLASS aud net/minecraft/village/VillagerProfession
+	FIELD a NONE Laud;
```
Comment (Gegy) [inline mappings/net/minecraft/village/VillagerProfession.mapping]: These too

File: mappings/net/minecraft/entity/SpawnType.mapping [hunk 9]
```diff
@@ -1 +0,0 @@
-CLASS aii net/minecraft/entity/SpawnType
```
Comment (Gegy) [inline mappings/net/minecraft/entity/SpawnType.mapping]: Just the variants need to be deleted -- not the whole mapping :D

Comment (KaptainWutax) [reply mappings/net/minecraft/entity/SpawnType.mapping]: Oops, didn't notice there was other stuff there...

Comment (KaptainWutax) [reply mappings/net/minecraft/entity/SpawnType.mapping]: Alright, fixed it.

**Comment by Gegy** [comment]
Hm, so we need to figure out a strategy in regards to the 3 way intersection with #527, #536 and #537

**Comment by KaptainWutax** [comment]
> Hm, so we need to figure out a strategy in regards to the 3 way intersection with #527, #536 and #537

#537 seems pretty easy to fix. I can't see any collisions with #527 though?

**Comment by Gegy** [comment]
#527 maps villager gossip related code

File: mappings/net/minecraft/entity/Entity.mapping [hunk 10]
```diff
@@ -80,6 +81,8 @@ CLASS ahw net/minecraft/entity/Entity
 	METHOD W getDataTracker ()Lqh;
 	METHOD Y invalidate ()V
 	METHOD Z getPose ()Laip;
+	METHOD a addParticles (B)V
+		ARG 1 particlesID
```
Comment (YanisBft) [inline mappings/net/minecraft/entity/Entity.mapping:85]: particlesId

**Pull request closed without merging**